### ISSUE TITLE
[Orbit] Cannot nest Orbit in an Accordion.

### ISSUE THREAD

**Issue by jellisii** [zurb/foundation-sites#9638]
Please note that this may be an Accordion bug.  

#### How to reproduce this bug:

1. Build out an accordion with a panel that has an orbit control on it.  
2. Attempt to activate the accordion tab containing the orbit control.

#### What should happen:
The orbit control should appear at an appropriate height.

#### What happened instead:
The panel does not expand to full height of the orbit container.

#### Test case:
https://codepen.io/jellisii/pen/bgpPmw

**Comment by jellisii** [comment]
Destroying and reinit-ing Orbit seems to sidestep the problem.  I'm unclear as to why orbit's `_reset` method does not seem to work.
```javascript
$('.orbit-accordion').on('down.zf.accordion', function() {
        var orbit = $(this).find('.accordion-item.is-active .orbit');
        $(orbit).foundation('destroy');
        // need to unhide the element
        $(orbit).show();
        $(orbit).foundation();
    });
```

**Comment by jellisii** [comment]
If there's an easy way to reflow the accordion panel after the orbit slide change, or if the accordion can pay attention to its content and reflow automagically on a change, this would work as I envision.

**Comment by ncoden** [comment]
Need some Js pro here :)
@zurbchris Would you have some time to take a look at it ?

> automagically

Love that word !

**Comment by kball** [comment]
@coreysyms is this a good usecase for `[data-mutate]`?

**Comment by jellisii** [comment]
I don't have a clean fix for this, but I did come up with a workaround that will at least get the spirit of what I'm trying to do done.  https://codepen.io/jellisii/pen/pRbVWV

From where I sit, it looks like Orbit doesn't like to behave when it's hidden, and doesn't reflow properly when it's shown after being hidden?  Maybe a foundation reInit() would be in order?  I haven't tested that yet.

**Comment by coreysyms** [comment]
Yes. I fire a  mutate event in tabs. I assume it would be similar for accordion. 

So I would need to set it up in both accordion and then a event listener for orbit I'll set it up  and see what happens locally

**Comment by coreysyms** [comment]
As a note. Orbit does not initialize in any div that is display none so targeting any method of that orbit instance does not work because it was never instantiated.

**Comment by jellisii** [comment]
I have a working implementation of orbit in accordion.  It's clunky, but I'm unsure how to use the mutation observer properly as of yet.  

On initial page render, set all accordion items with `.is-active`.  Have a method of flagging which items you want to stay open.  I'm using `.default-expanded`.  This will render the tab open and allow orbit to init properly when you fire `$(document).foundation()`.  

The following listens for down events on a specified container, and sets fires `up` on the items not having `.default-expanded` after all the intial down events have fired.

```javascript
var timeoutContainer;
$('#accordion_with_orbit_elements').on('down.zf.accordion', function() {
    if (timeoutContainer !== undefined) {
        clearTimeout(timeoutContainer);
    }
    var this_accordion = this;
    timeoutContainer = setTimeout(function() {
        $(this_accordion).find('.accordion-item.is-active').each(function() {
            if (!$(this).hasClass('default-expanded')) {
                $(this_accordion).foundation('up', $(this).find('.accordion-content'));
            }
        });
        $(this_accordion).off('down.zf.accordion');
    }, 250);
});
```

I'm completely confident that @coreysyms probably has a much better solution though. :)

**Comment by DanielRuf** [comment]
> As a note. Orbit does not initialize in any div that is display none so targeting any method of that orbit instance does not work because it was never instantiated.

Right. How should we handle it then? I guess this is still unsolved. So far we initialize uninitialized components when the parent is opened (reveal, accordion, tabs) or use Swiper with its observer settings.

**Comment by danlevinA6** [comment]
So I ran into a similar issue when placing an orbit into a tab that was not visible. When you clicked to the tab the orbit-container would have a "height=0" in its inline style. The fix I used and could be easily applied to Accordions is triggering a window resize on tab change. This forced a resize to the orbit container. I am sure there is a better way to handle this but this works. 
`$( document ).ready(function() {
	    console.log( "tabs slider ready!" );
	   	
	   	$('[data-tabs]').on('change.zf.tabs', function() {
	   	  console.log('Those tabs sure did change!');
	   	  $(window).trigger('resize');
	   	});
	});`

**Comment by mcoogan62** [comment]
Could you let me know why this issue is closed according to joeworkman? This flat-out does not work. I have no problems using a carousel inside an accordion with Bootstrap 5.
